src/ceph-volume/ceph_volume/util/device.py

```diff
@@ -153,7 +153,12 @@ def _parse(self):
                 if part:
                     self.sys_api = part
                     break
-
+        if not self.sys_api:
```

Comment:
there's already the same condition L148

Reply:
Right, I didn't want to check for symlink if partition was already matched at L154
```
self.sys_api = part
break
```
What do you propose?

Reply:
hey sorry for the late response.
a bit busy at the moment but if you aren't in a hurry I plan to take a look at this next week @jansobczak

Reply:
hey @jansobczak, I've started to take a look at this, I'll get back to you by the end of the week

Reply:
@jansobczak what about something like following:
```diff
diff --git a/src/ceph-volume/ceph_volume/util/device.py b/src/ceph-volume/ceph_volume/util/device.py
index 193a72f3d00..73ae9b8f16e 100644
--- a/src/ceph-volume/ceph_volume/util/device.py
+++ b/src/ceph-volume/ceph_volume/util/device.py
@@ -144,6 +144,10 @@ class Device(object):
     def _parse(self):
         if not sys_info.devices:
             sys_info.devices = disk.get_devices()
+        # check if we are a symlink
+        if os.path.islink(self.abspath):
+            self.abspath = os.path.join(os.path.dirname(self.abspath),
+                                    os.readlink(self.abspath))
         self.sys_api = sys_info.devices.get(self.abspath, {})
         if not self.sys_api:
             # if no device was found check if we are a partition
```

```
[vagrant@ceph ~]$ ls -l /dev/foo*
lrwxrwxrwx. 1 root root 8 Dec  2 14:03 /dev/fooA -> /dev/sda
lrwxrwxrwx. 1 root root 8 Dec  2 14:03 /dev/fooB -> /dev/sdb
lrwxrwxrwx. 1 root root 8 Dec  2 14:03 /dev/fooC -> /dev/sdc
[vagrant@ceph ~]$ sudo /usr/local/bin/ceph-volume lvm batch --filestore /dev/fooA /dev/fooB --journal-devices /dev/fooC --report
--> passed data devices: 2 physical, 0 LVM
--> relative data size: 1.0
--> passed journal devices: 1 physical, 0 LVM

Total OSDs: 2

  Type            Path                                                    LV Size         % of device
----------------------------------------------------------------------------------------------------
  data            /dev/fooA                                               10.74 GB        100.00%
  journal         /dev/fooC                                               5.37 GB         50.00%
----------------------------------------------------------------------------------------------------
  data            /dev/fooB                                               10.74 GB        100.00%
  journal         /dev/fooC                                               5.37 GB         50.00%
[vagrant@ceph ~]$ sudo /usr/local/bin/ceph-volume lvm batch --bluestore  /dev/fooA /dev/fooB --db-devices /dev/fooC --report
--> passed data devices: 2 physical, 0 LVM
--> relative data size: 1.0
--> passed block_db devices: 1 physical, 0 LVM

Total OSDs: 2

  Type            Path                                                    LV Size         % of device
----------------------------------------------------------------------------------------------------
  data            /dev/fooA                                               10.74 GB        100.00%
  block_db        /dev/fooC                                               5.37 GB         50.00%
----------------------------------------------------------------------------------------------------
  data            /dev/fooB                                               10.74 GB        100.00%
  block_db        /dev/fooC                                               5.37 GB         50.00%
[vagrant@ceph ~]$ sudo /usr/local/bin/ceph-volume lvm batch --bluestore  /dev/fooA --db-devices /dev/fooB --wal-devices /dev/fooC --report
--> passed data devices: 1 physical, 0 LVM
--> relative data size: 1.0
--> passed block_db devices: 1 physical, 0 LVM
--> passed block_wal devices: 1 physical, 0 LVM

Total OSDs: 1

  Type            Path                                                    LV Size         % of device
----------------------------------------------------------------------------------------------------
  data            /dev/fooA                                               10.74 GB        100.00%
  block_db        /dev/fooB                                               10.74 GB        100.00%
  block_wal       /dev/fooC                                               10.74 GB        100.00%
[vagrant@ceph ~]$ 
```

at first glance, it seems to work, but it might require more work (and testing).

I have still a concern, I wonder whether this is a nice idea to support this, I'm not that confident when it comes to playing with symlinks. I'm just thinking about this off the top of my head.

@jan--f  any thoughts?

Reply:
This is fine and it's even better cause line with ```         self.sys_api = sys_info.devices.get(self.abspath, {})``` isn't duplicated. 
As for symlinks and is it nice to support it. The main reason we would like to use udev rules is sometimes SAS controllers change their SCSI index which results in different naming of disks. Consider server A 
```
/dev/sda -> system (raid1)
/dev/sdb -> system (raid1)

/dev/sdc -> OSD
/dev/sdd -> OSD
/dev/sde -> OSD
```
And then server B which is identical to server A considering all hardware components but have different SCSI index for SAS controller and we end up with something like:
```
/dev/sda -> OSD
/dev/sdb -> OSD
... 
/dev/sdx  -> system (raid1)
/dev/sdz  -> system (raid1)
```
We use standard hardware configuration across the cluster so udev rules fix this problem for us but we were unable to provision OSD when we passed symlinks to ceph-volume.